app/src/main/java/com/kamron/pogoiv/MainActivity.java

```diff
@@ -2,6 +2,7 @@
 
 import android.Manifest;
 import android.annotation.TargetApi;
```

Comment:
As @Blaisorblade said, my "split into 4 commits" should not be taken to be an exact request you have to follow blindly. It's just seems too big to be one commit. Maybe you can only split it into 2 or 3 commits if you are to meet the other requirements @Blaisorblade mentioned (like having to compile and not be completely broken, etc).

I personally would do exactly what @Blaisorblade said, but if you are new to git, stash and "rebase -i" could be very disorienting. In which case, I'd recommend making a temp branch that's a copy of this. Then reset this branch to master and cherry-pick the commits one at a time from the temp branch. Once you cherry-pick a commit and it's the latest commit in your current branch, you can split it up by doing:

1. git reset HEAD~1 (this will remove the commit from log, but will leave the changes on files)
2. git diff (this should show that the above worked)
3. git add -p (this show pieces of code changes close to each other and asks you if you want to add it or not. Only add the ones you want to go into split commit 1)
4. git commit (fill up the test and commit it)
5. go to step 3 to finish split commit 2, 3, 4...

Hope this helps.